agenta-backend/agenta_backend/routers/container_router.py

```diff
@@ -163,8 +164,20 @@ async def construct_app_container_url(
 
     if base_id:
         object_db = await db_manager.fetch_base_by_id(base_id)
-    elif variant_id:
+    elif variant_id and variant_id != "None":
+        # NOTE: Backward Compatibility
+        # ---------------------------
```

Comment:
why "backward compatibility" ?
this happens with new apps as well

Reply:
No, it will not. Because we are now ensuring that the user cannot delete a variant until they're removed the resource(s) that _heavily_ depends on it.

Reply:
The code you made a comment on is only for users that have deleted a variant or variants despite having evaluations tied to that variant.

Reply:
what do you mean no it will not ? it will not happen with new apps?
I was able to reproduce it many times with new apps

Reply:
[Conversation resolved]

Question replied in a call session in Google meet.